server/routes/prisonerReport/prisonerReportPage.ts

```diff
@@ -162,14 +176,39 @@ export default class prisonerReportRoutes {
   }
 
   submit = async (req: Request, res: Response): Promise<void> => {
+    const adjudicationNumber = Number(req.params.adjudicationNumber)
+    const { user } = res.locals
+
     const status = this.reviewStatus(req.body.currentStatusSelected)
     const reason = this.reason(status, req)
     const details = this.details(status, req)
 
     const error = validateForm({ status, reason, details })
     if (error) return this.renderView(req, res, { errors: [error], status, reason, details })
 
-    return res.redirect(adjudicationUrls.allCompletedReports.root)
+    const url = req.query.referrer as string
+    const urlStatus = req.query.status as string
+    const urlToDate = req.query.toDate as string
+
+    try {
+      await this.reportedAdjudicationsService.reviewAdjudication(adjudicationNumber, status, reason, details, user)
+      return res.redirect(`${url}&toDate=${urlToDate}&status=${urlStatus}`)
+    } catch (e) {
+      if (e.status === 400) {
```

Comment:
Hmm, yes I see that 400 is thrown if the user attempts an invalid transition - but is there anything they can do about it?

This is why we generally assume any API error cannot be resolved by the user, and therefore we don't tell them about it (we show a "sorry" page)

Reply:
removed